src/lexer/scan.rs

```diff
@@ -53,6 +53,38 @@ impl Input for &[u8] {
     }
 }
 
+impl Input for Vec<u8> {
+    #[inline]
+    fn fill_buf(&mut self) -> io::Result<()> {
+        Ok(())
+    }
+
+    #[inline]
+    fn eof(&self) -> bool {
+        true
+    }
+
+    #[inline]
+    fn consume(&mut self, amt: usize) {
+        self.drain(..amt);
```

Comment:
This seems wrong to me:
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain